httputil/client.go

```diff
@@ -0,0 +1,58 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package httputil
+
+import (
+	"crypto/tls"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+type ClientOpts struct {
```

Comment:
It's a mixed bag, but I think we generally expand "Options" more than we have "Opts":
```
[niemeyer@nomad ../snapcore/snapd]% grep 'Opts{' * -r | wc -l
30

[niemeyer@nomad ../snapcore/snapd]% grep 'Options{' * -r | wc -l
246
```

Reply:
Thanks, I did a followup for the renames https://github.com/snapcore/snapd/pull/5729

Reply:
Thanks!